experiment/Commonfunctions/commonfunctions-practice.js

```diff
@@ -215,48 +223,11 @@ function visitElement(element,animX){
 
 
 var animX1=0;
-function recursiveInorder(root){ 
-  if(root!==undefined){
-    if(root.children!==undefined)
-      recursiveInorder(root.children[0])
-   tree_traversal.sequence_list.push(root.name);
-   console.log("called",tree_traversal.sequence_list);
-    
-    if(root.children!==undefined)
-      recursiveInorder(root.children[1])
-  }
-}
-
-function recursivePreorder(root)
-{
-  console.log("called",root)
-  if(root!==undefined)
-  {
-    tree_traversal.sequence_list.push(root.name);
-    if(root.children!==undefined)
-      recursivePreorder(root.children[0])
-    if(root.children!==undefined)
-      recursivePreorder(root.children[1])
-  }
-}
-function recursivePostorder(root)
-{
-  console.log("called",root)
-  if(root!==undefined)
-  {  
-    if(root.children!==undefined)
-      recursivePostorder(root.children[0]) 
-    if(root.children!==undefined)
-      recursivePostorder(root.children[1])
-     tree_traversal.sequence_list.push(root.name)
-  }
-}
-
-
 
 function generateG(){
   resetTraversal();
   generateGraph();
 }
+window.generateG=generateG;
```

Comment:
I understand, the issue is the resetTraversal that is being called in generateG is different for practice and exercise. This was one of the major hurdles I ran into when trying to modularize. I tried making them the  same but the way the code is written is too rigid for me to be able to do it. I would have to re-write practice and demo so that they operate the same way which would be extremely tedious work.